driver/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala

```diff
@@ -768,6 +769,29 @@ object TypeConverter {
     }
   }
 
+  class CqlVectorConverter[T <: Number : TypeConverter](dimension: Int) extends TypeConverter[CqlVector[T]] {
+    val elemConverter = implicitly[TypeConverter[T]]
+
+    implicit def elemTypeTag: TypeTag[T] = elemConverter.targetTypeTag
+
+    @transient
+    lazy val targetTypeTag = {
+      implicitly[TypeTag[CqlVector[T]]]
+    }
+
+    def newCollection(items: Iterable[Any]): java.util.List[T] = {
+      val buf = new java.util.ArrayList[T](dimension)
```

Comment:
Why java collection here?

Reply:
This collection is passed to `CqlVector.getInstance` which expects either Java `List` or array. If we use here a Scala collection and then let `JavaConverters` adjust that, we will have to create one extra wrapper which is a waste in this case.